code/datums/components/sign_language.dm

```diff
@@ -255,29 +274,51 @@
 	var/exclamation_found = findtext(message, "!")
 	// Is there a ?
 	var/question_found = findtext(message, "?")
+	var/emote_tone = TONE_NEUTRAL
+	if (exclamation_found && question_found)
+		emote_tone = TONE_INQUISITIVE_EMPHATIC
+	else if (exclamation_found)
+		emote_tone = TONE_EMPHATIC
+	else if (question_found)
+		emote_tone = TONE_INQUISITIVE
 
 	// Cut our last overlay before we replace it
 	if(timeleft(tonal_timerid) > 0)
 		remove_tonal_indicator()
 		deltimer(tonal_timerid)
-	// Prioritize questions
-	if(question_found)
-		tonal_indicator = mutable_appearance('icons/mob/effects/talk.dmi', "signlang1", TYPING_LAYER)
-		carbon_parent.visible_message(span_notice("[carbon_parent] lowers [carbon_parent.p_their()] eyebrows."))
-	else if(exclamation_found)
-		tonal_indicator = mutable_appearance('icons/mob/effects/talk.dmi', "signlang2", TYPING_LAYER)
-		carbon_parent.visible_message(span_notice("[carbon_parent] raises [carbon_parent.p_their()] eyebrows."))
-	// If either an exclamation or question are found
+	// Determine tone based on punctuation
```

Comment:
```suggestion
```

this isn't the part where we determine tone, we already did that six lines up. this can live without a comment